pkg/deviceshifu/deviceshifuopcua/deviceshifuopcua.go

```diff
@@ -50,6 +52,15 @@ func New(deviceShifuMetadata *deviceshifubase.DeviceShifuMetaData) (*DeviceShifu
 	if err != nil {
 		return nil, fmt.Errorf("Error parsing ConfigMap at %v", deviceShifuMetadata.ConfigFilePath)
 	}
+
+	setting := *base.EdgeDevice.Spec.ProtocolSettings.OPCUASetting
+	passwordByte, err := os.ReadFile(DeviceSecretPasswordPath)
+	password := *setting.Password
+	// secret will overwrite the password in edge device
+	if err == nil {
```

Comment:
what if err != nil?

Reply:
My idea is to try not to modify the original password loading logic, and try to load the password from the secret and overwrite the original plaintext password. If there is any error in the process of loading the secret, the original method will still be used.

Reply:
ok, I think at least we can't swallow the error message here. 
Because some time if you want to know why the password is not used from secret file, there would be no clue in the log.

would a line of error log make this clear?

Reply:
ok, now added some logs